components/workflow.rst

```diff
@@ -19,6 +19,117 @@ You can install the component in 2 different ways:
+
+Usage
+-----
+
+The workflow component gives you an object oriented way to work with state machines. A state machine lets you
+define *places* (or *states*) and *transactions*. A transaction describes the action to get from one place to another.
+
+.. image:: /_images/components/workflow/states_transactions.png
+
+A set of places and ``Transaction's`` creates a ``Definition``. A ``Workflow`` needs a ``Definition`` and a way to write
+the states to the objects, ie an instance of a ``MarkingStoreInterface``.
+
+Consider the following example for a blog post. A post can have places: 'draft', 'review', 'rejected', 'published'. You
+can define the workflow like this::
+
+    $states = ['draft', 'review', 'rejected', 'published'];
+    $transactions[] = new Transition('to_review', ['draft', 'rejected'], 'review');
+    $transactions[] = new Transition('publish', 'review', 'published');
+    $transactions[] = new Transition('reject', 'review', 'rejected');
+
+    $definition = new Definition($states, $transactions);
+    $definition->setInitialPlace('draft');
+
+    $marking = new ScalarMarkingStore('currentState');
+    $workflow = new Workflow($definition, $marking);
+
+The ``Workflow`` can now help you to decide what actions that are allowed on a blog post.
+
+.. code-block:: php
+
+    $post = new \stdClass();
+    $post->currentState = null;
+    $workflow->can($post, 'publish'); // False
+    $workflow->can($post, 'to_draft'); // True
+
+    // Update the currentState on the post
+    try {
+        $workflow->apply($post, 'to_review');
+    } catch (LogicException $e) {
+        // ...
+    }
+
+    // See all the available transaction for the post in the current state
+    $transactions = $workflow->getEnabledTransitions($post);
+
+
+
+Using events
+------------
+
+To make your workflows even more powerful you could construct the ``Workflow`` object with an ``EventDispatcher``. You
+can now create event listeners to block transactions ie depending on the data in the blog post. The following events
+are dispatched:
+
+* workflow.guard
+* workflow.[workflow name].guard
+* workflow.[workflow name].guard.[transaction name]
+
+See example to make sure no blog post without title is moved to "review"::
+
+    $marking = new ScalarMarkingStore('currentState');
+    $workflow = new Workflow($definition, $marking, $dispatcher, 'blogpost');
+
+.. code-block:: php
+
+    class BlogPostReviewListener implements EventSubscriberInterface
+    {
+        public function guardReview(GuardEvent $event)
+        {
+            $post = $event->getSubject();
+            $title = $post->getTitle();
+
+            if (empty($title)) {
+                // Posts with no title should not be allowed
+                $event->setBlocked(true);
+            }
+        }
+
+        public static function getSubscribedEvents()
+        {
+            return array(
+                'workflow.blogpost.guad.to_review' => array('guardReview'),
```

Comment:
I guess there is a typo here, isn't it ? 

```diff
-  'workflow.blogpost.guad.to_review'
+ 'workflow.blogpost.guard.to_review'
```
Is it related to the ``Guard`` component ? I don't see any security stuff here.